mkdir_linux.go

```diff
@@ -185,8 +188,16 @@ func MkdirAllHandle(root *os.File, unsafePath string, mode int) (_ *os.File, Err
 		if stat, err := fstat(currentDir); err != nil {
 			return nil, fmt.Errorf("check newly created directory: %w", err)
 		} else {
-			if stat.Mode != expectedMode {
-				return nil, fmt.Errorf("%w: newly created directory %q has incorrect mode 0o%.3o (expected 0o%.3o)", errPossibleAttack, currentDir.Name(), stat.Mode, expectedMode)
+			if stat.Mode&^maxExpectedMode != 0 {
+				// Unfortunately, we can't return an error. Filesystems like
+				// vfat don't support Unix-like DAC modes, and we have had
+				// spurious errors from this check a few times so we should
+				// play it safe to avoid regressions. The uid/gid check is more
+				// important and to be honest these checks are not that
+				// important (they mostly exist out of an abundance of caution
+				// -- it's not entirely clear what practical attack this is
+				// defending against).
+				logrus.Printf("filepath-securejoin: %v: newly created directory %q has incorrect mode 0o%.3o (expected 0o%.3o)", errPossibleAttack, currentDir.Name(), stat.Mode, maxExpectedMode)
```

Comment:
nit: perhaps it's better to use `logrus.Infof` (to be explicit about the log level -- otherwise Printf is the same as Infof).

Reply:
Ah, this is from when I used the stdlib `log`. Fixed.